brightener.cpp

```diff
@@ -1,32 +1,63 @@
 #include "brightener.h"
 
+Image::Image(int r, int c) : rows(r), columns(c) {
+    // Validate dimensions
+    if (r <= 0 || c <= 0) {
+        throw std::runtime_error("Invalid image dimensions.");
+    }
+
+    // Check for potential overflow
+    if (static_cast<long long>(r) * static_cast<long long>(c) > std::numeric_limits<size_t>::max()) {
+        throw std::runtime_error("Image size exceeds maximum allowable size.");
+    }
+
+    // Allocate memory for pixels using unique_ptr
+    pixels = std::make_unique<uint8_t[]>(r * c);
+}
+
+// Move constructor
+Image::Image(Image&& other) noexcept
+    : rows(other.rows), columns(other.columns), pixels(std::move(other.pixels)) {
+    other.rows = 0;   // Optional: reset the moved-from object's members
```

Comment:
Why does the comment say "Optional"? A move-constructor is meant to remove ownership from the rvalue, right?